src/preprocessing/passes/ackermann.cpp

```diff
@@ -187,9 +188,110 @@ void collectFunctionsAndLemmas(FunctionToArgsMap& fun_to_args,
+ * new BV type */
+size_t getBVSkolemSize(size_t capacity)
+{
+  return static_cast<size_t>(log2(capacity)) + 1;
+}
+
+/* Given the lowest capacity requirements for each uninterpreted sort, assign
+ * a sufficient bit-vector size.
+ * Get the map from variables with uninterpreted sort to the fresh skolem BV
+ * variables */
+void collectUSortsToBV(const unordered_set<TNode, TNodeHashFunction>& vars,
+                       USortToBVSizeMap& usortCardinality,
+                       SubstitutionMap& usVarsToBVVars)
+{
+  NodeManager* nm = NodeManager::currentNM();
+
+  for (TNode var : vars)
+  {
+    TypeNode type = var.getType();
+    size_t size = getBVSkolemSize(usortCardinality[type]);
+    Node skolem = nm->mkSkolem(
+        "BVSKOLEM$$",
+        nm->mkBitVectorType(size),
+        "a variable created by the ackermannization "
+        "preprocessing pass, representing a variable with uninterpreted sort "
+            + type.toString() + ".");
+    usVarsToBVVars.addSubstitution(var, skolem);
+  }
+}
+
+/* This function returns the list of terms with uninterpreted sort in the
+ * formula represented by assertions. */
+std::unordered_set<TNode, TNodeHashFunction> getVarsWithUSorts(
+    AssertionPipeline* assertions)
+{
+  std::unordered_set<TNode, TNodeHashFunction> res;
+
+  for (const Node& assertion : assertions->ref())
+  {
+    std::unordered_set<TNode, TNodeHashFunction> vars;
+    expr::getVariables(assertion, vars);
+
+    for (const TNode& var : vars)
+    {
+      if (var.getType().isSort())
+      {
+        res.insert(var);
+      }
+    }
+  }
+
+  return res;
+}
+
+/* This is the top level of converting uninterpreted sorts to bit-vectors.
+ * We count the number of different variables for each uninterpreted sort.
+ * Then for each sort, we will assign a new bit-vector type with a sufficient
+ * size. The size is calculated to have enough capacity, that can accommodate
+ * the variables occured in the original formula. At the end, all variables of
+ * uninterpreted sorts will be converted into Skolem variables of BV */
+void usortsToBitVectors(const LogicInfo& d_logic,
+                        AssertionPipeline* assertions,
+                        USortToBVSizeMap& usortCardinality,
+                        SubstitutionMap& usVarsToBVVars)
+{
+  std::unordered_set<TNode, TNodeHashFunction> toProcess =
+      getVarsWithUSorts(assertions);
+
+  if (toProcess.size() > 0)
+  {
+    /* the current version only supports BV for removing uninterpreted sorts */
+    AlwaysAssert(d_logic.isTheoryEnabled(theory::THEORY_BV))
+        << "Cannot use Ackermannization on formula with uninterpreted "
+           "sorts without BV logic";
+
+    for (TNode term : toProcess)
+    {
+      TypeNode type = term.getType();
+      /* Update the statistics for each uninterpreted sort.
+       * For non-existing keys, C++ will create a new element for it, which has
+       * the value initialized with a pair of two zeros. */
```

Comment:
Is it still a pair of two zeros? I think now it is only a single zero.
And actually right there you increment it by 1, and then the result is 1, not 0. so I suggest:

```suggestion
       * a default 0 value, before incrementing by 1. */
```